src/Storages/MergeTree/PartMetadataManagerWithCache.cpp

```diff
@@ -117,29 +117,45 @@ void PartMetadataManagerWithCache::updateAll(bool include_projection)
 
     String value;
     String read_value;
-    for (const auto & file_name : file_names)
-    {
-        String file_path = fs::path(part->getDataPartStorage().getRelativePath()) / file_name;
-        if (!part->getDataPartStorage().exists(file_name))
-            continue;
-        auto in = part->getDataPartStorage().readFile(file_name, {}, std::nullopt, std::nullopt);
-        readStringUntilEOF(value, *in);
 
-        String key = getKeyFromFilePath(file_path);
-        auto status = cache->put(key, value);
-        if (!status.ok())
+    /// This is used to remove the keys in case of any exception while caching other keys
+    Strings keys_added_to_cache(file_names.size());
```

Comment:
afair we should create it empty and call .reserve() instead of argument in constructor. Current version will create vector of empty strings.

Reply:
sorry my bad. addressed it